pyrdp/parser/rdp/fastpath.py

```diff
@@ -368,27 +361,24 @@ def isCompressed(self, header: int) -> bool:
         return (header >> 6) & FastPathOutputCompressionType.FASTPATH_OUTPUT_COMPRESSION_USED != 0
 
     def parse(self, data: bytes) -> FastPathOutputEvent:
+        """
+        Parse TS_FP_UPDATE.
+
+        https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/a1c4caa8-00ed-45bb-a06e-5177473766d3
+        """
         stream = BytesIO(data)
         header = Uint8.unpack(stream)
 
-        compressionFlags = None
-
-        if self.isCompressed(header):
-            compressionFlags = Uint8.unpack(stream)
-
+        # https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/a1c4caa8-00ed-45bb-a06e-5177473766d3
+        updateCode = header & 0xf
+        # fragmentation = (header & 0b00110000) >> 4
+        compressionFlags = Uint8.unpack(stream) if self.isCompressed(header) else None
         size = Uint16LE.unpack(stream)
 
-        eventType = header & 0xf
-        fragmentation = header & 0b00110000 != 0
-
-        if fragmentation:
-            log.debug("Fragmentation is present in output fastpath event packets."
-                      " Not parsing it and saving to FastPathOutputUpdateEvent.")
-            return FastPathOutputEvent(header, compressionFlags, payload=stream.read(size))
```

Comment:
Nothing, the function will parse the first fragment of the order, which will create the proper type in Python, which is used down the line where the actual reassembly of fragments happens (In the Player code). It actually simplifies logic further down the pipeline since we don't have to re-check and re-parse the FastPathOutputType as a specific type a second time.

Reply:
Awesome. Resolving.